### ISSUE TITLE
Facing NaN issue in train_second.py

### ISSUE THREAD

**Issue by thenameisAnurag** [yl4579/StyleTTS2#347]
i have successfully trained the mode on train_first.py 

this was the result in  the train.logs 
```
INFO:2025-09-25 17:50:53,200: Epoch [50/50], Step [240/243], Mel Loss: 0.28113, Gen Loss: 0.00000, Disc Loss: 0.00000, Mono Loss: 0.00000, S2S Loss: 0.00000, SLM Loss: 0.00000
INFO:2025-09-25 17:51:06,578: Validation loss: 0.275
```
now i have my config file with 
`epochs_1st: 0
epochs_2nd: 60             
batch_size: 2
max_len: 500`

and now i am facing this issue 
`INFO:2025-09-30 15:24:19,919: Epoch [47/60], Step [20/486], Loss: nan, Disc Loss: nan, Dur Loss: nan, CE Loss: nan, Norm Loss: nan, F0 Loss: nan, LM Loss: nan, Gen Loss: nan, Sty Loss: nan, Diff Loss: nan, DiscLM Loss: 0.00000, GenLM Loss: 0.00000
INFO:2025-09-30 15:42:13,051: Epoch [47/60], Step [10/486], Loss: nan, Disc Loss: nan, Dur Loss: nan, CE Loss: nan, Norm Loss: nan, F0 Loss: nan, LM Loss: nan, Gen Loss: nan, Sty Loss: nan, Diff Loss: nan, DiscLM Loss: 0.00000, GenLM Loss: 0.00000

please suggest and help how can i solve this issue 

i have 
removed the 
`set_trace()  `

but still facing the issue what i need to do

**Comment by patriotyk** [comment]
You didn't train `train_first.py` properly because from your logs I see that GAN train(TMA_epoch parameter in your config seems to be greater or equals to epochs_1st) has newer started. Also it is possible to train to 0.24 validation loss before GAN train started. I recommend to increase epochs_1st to 200 and set TMA_epoch то 80. Also don't be confused when TMA_epoch start, mel loss should quickly go up then slowly begin decreasing.

**Comment by thenameisAnurag** [comment]
> You didn't train `train_first.py` properly because from your logs I see that GAN train(TMA_epoch parameter in your config seems to be greater or equals to epochs_1st) has newer started. Also it is possible to train to 0.24 validation loss before GAN train started. I recommend to increase epochs_1st to 200 and set TMA_epoch то 80. Also don't be confused when TMA_epoch start, mel loss should quickly go up then slowly begin decreasing.

Thank you bro , i will implement and check the results
